src/nccl.h.in

```diff
@@ -39,7 +39,10 @@ typedef struct ncclComm* ncclComm_t;
 #define NCCL_UNIQUE_ID_BYTES 128
 /*! @brief      Opaque unique id used to initialize communicators
     @details    The ncclUniqueId must be passed to all participating ranks */
-typedef struct { char internal[NCCL_UNIQUE_ID_BYTES]; /*!< Opaque array>*/} ncclUniqueId;
+typedef struct {
+    char internal[NCCL_UNIQUE_ID_BYTES]; /*!< Opaque array>*/
+    char internal2[NCCL_UNIQUE_ID_BYTES]; /*!< Opaque array>*/
```

Comment:
Should this be `ifdef`ed by `ENABLE_MSCCLPP` or some other flag?

Reply:
The problem is that the client code must also define it.

Reply:
After talking to @wenkaidu we feel that we need to do this without changing the NCCL API. He suggested I try to distribute the unique ID internally using rccl's bootstrap allgather.